### PR TITLE
[SQL] `az sql db ledger-digest-uploads`:  Support SQL Ledger

### PR DESCRIPTION
**Pull request by rewongmicrosoft**
**Description**

Adding Azure CLI commands for SQL Ledger (az sql db ledger-digest-uploads). 
Adding parameter to az sql db create for ledger database 

See issue here: https://github.com/Azure/azure-cli/issues/18339

**Testing Guide**

enables ledger digest uploading to endpoint and database specified
az sql db ledger-digest-uploads enable --endpoint
[--ids]
[--name]
[--resource-group]
[--server]
[--subscription]

disables ledger digest uploading to database specified
az sql db ledger-digest-uploads disable 
[--ids]
[--name]
[--resource-group]
[--server]
[--subscription]

shows configured ledger digest upload for database specified 
az sql db ledger-digest-uploads show 
[--ids]
[--name]
[--resource-group]
[--server]
[--subscription]

creates a az sql db with ledger enabled
az sql db create -g mygroup -s myserver -n mydb -ledger-on

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
SQL

File: src/azure-cli/azure/cli/command_modules/sql/_format.py [hunk 1]
```diff
@@ -152,6 +152,10 @@ def db_transform(result):
     result.edition = result.sku.tier
     result.elastic_pool_name = _last_segment(result.elastic_pool_id)
 
+    # rename ledger setting name
+    result.ledger_on = result.is_ledger_on
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_format.py:156]: what about adding check for is_ledger_on existence?

File: src/azure-cli/azure/cli/command_modules/sql/_help.py [hunk 2]
```diff
@@ -1523,3 +1523,32 @@
...
 """
+
+helps['sql db ledger-digest-uploads'] = """
+type: group
+short-summary: Manage ledger digest upload settings.
+"""
+
+helps['sql db ledger-digest-uploads enable'] = """
+type: command
+short-summary: Enables uploading ledger digests to an Azure Storage account or to Azure Confidential Ledger. If uploading ledger digests is already enabled, the cmdlet resets the digest storage endpoint to a new value.
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_help.py:1534]:
```suggestion
short-summary: Enable uploading ledger digests to an Azure Storage account or to Azure Confidential Ledger. If uploading ledger digests is already enabled, the cmdlet resets the digest storage endpoint to a new value.
```

File: src/azure-cli/azure/cli/command_modules/sql/_help.py [hunk 3]
```diff
@@ -1523,3 +1523,32 @@
...
+examples:
+  - name: Enable uploading ledger digests to an Azure Blob storage.
+    text: az sql db ledger-digest-uploads enable --name mydb --resource-group MyResourceGroup --server myserver --endpoint https://mystorage.blob.core.windows.net
+"""
+
+helps['sql db ledger-digest-uploads disable'] = """
+type: command
+short-summary: Disables uploading ledger digests.
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_help.py:1542]:
```suggestion
short-summary: Disable uploading ledger digests.
```

File: src/azure-cli/azure/cli/command_modules/sql/_params.py [hunk 4]
```diff
@@ -211,6 +211,13 @@ def get_location_type_with_default_from_resource_group(cli_ctx):
     options_list=['--maint-config-id', '-m'],
     help='Specified maintenance configuration id or name for this resource.')
 
+ledger_on_param_type = CLIArgumentType(
+    options_list=['--ledger-on'],
+    help='Creates a ledger database, in which the integrity of all data is protected by the ledger feature. '
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:216]:
```suggestion
    help='Create a ledger database, in which the integrity of all data is protected by the ledger feature. '
```

**Comment by rewongmicrosoft** [comment]
@Juliehzl updated based on your comments

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**